controller/src/main/java/org/jboss/as/controller/operations/validation/ObjectTypeValidator.java

```diff
@@ -51,14 +53,33 @@ public ObjectTypeValidator(final boolean nullable, final AttributeDefinition...
     public void validateParameter(final String parameterName, final ModelNode value) throws OperationFailedException {
         super.validateParameter(parameterName, value);
         if (value.isDefined()) {
+            ModelNode clonedValue = null;
+            StringBuilder allowed = new StringBuilder();
```

Comment:
I'd prefer this isn't declared until the https://github.com/wildfly/wildfly-core/pull/4292/files#diff-377f68b7791de46e877e8bc15afc390eR75 failure handling block is entered. Then in that block loop through the allowedValues.keys().

IOW the failure case will be an edge case so for the normal case we should do the minimal work related to it, even if that means a bit more work in the failure case.